src/celengine/selection.h

```diff
@@ -87,4 +96,9 @@ inline bool operator!=(const Selection& s0, const Selection& s1)
     return s0.type != s1.type || s0.obj != s1.obj;
 }
 
+inline bool operator<(const Selection& s0, const Selection& s1)
+{
+    return s0.type < s1.type || s0.obj < s1.obj;
```

Comment:
I'm not sure that you do what you actually want.
1. I suppose you should compare objects only if `s0.type == s1.type`
1. obj is a pointer, so pointer comparison is nonsense, do you meant `*s0.obj < *s1.obj`?

Reply:
Pointer comparison must not be used here. It's valid for `!=` to ensure that s0 and s1 doesn't point to the same object. But not in this case.

```
if (s0.type != s1.type)
    return s0.type < s1.type;

// here goes more complicated code to work with s0.type == s1.type
// not an actual code, but pseudo code
switch (s0.type)
{
case Type_Nil: return false;
case Type_Body: return *static_cast<Body*>s0.obj < *static_cast<Body*>s1.obj;
...
}
```

Reply:
Hmm, I see no problem. Pointers comparisions is standard procedure in c++ , as pointer is just unsigned integer. It is fast and reliable. I see also no ease workaround. (Casting to typed pointer changes nothing.) I used it mamy times with many kinds of containers with any problem.

Reply:
Hmm, I see no problem. Pointers comparisions is standard procedure in c++ , as pointer is just unsigned integer. It is fast and reliable. I see also no ease workaround. (Casting to typed pointer changes nothing.) I used it mamy times with many kinds of containers with any problem.

Reply:
Okay, let me explain. You have two planets, Earth and Saturn. Both are Type_Body, but Earth is located in memory at 0x0100 and Saturn is located at 0x0010. Your code says that Saturn < Earth. That's absurd.

Reply:
And this absurd is needed to store it in std::set. I hope nobody will try to compare two select objects, expecting anything else.

Reply:
Hi @pirogronian, I've just realized that you won't need this operator if you use `unordered_set` instead of `set`. It seams that you actually don't need ordering.

Reply:
Segfaults eliminated, code simplified.